File: CineBox/UI/cinebox_desktop/lib/models/Booking/booking.dart
import 'package:cinebox_desktop/models/Promotion/promotion.dart';
import 'package:cinebox_desktop/models/Screening/screening.dart';
import 'package:cinebox_desktop/models/Users/users.dart';
import 'package:json_annotation/json_annotation.dart';

part 'booking.g.dart';

@JsonSerializable()
class Booking {
  int? id;
  int? userId;
  int? screeningId;
  double? price;
  int? promotionId;
  Screening? screening;
  Promotion? promotion;
  Users? users;

  Booking(this.id, this.userId, this.screeningId, this.price, this.promotionId,
      this.screening, this.promotion, this.users);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory Booking.fromJson(Map<String, dynamic> json) =>
      _$BookingFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$BookingToJson(this);
}

File: CineBox/CineBox.Model/ViewModels/Booking.cs
using System;
namespace CineBox.Model.ViewModels
{
	public partial class Booking
	{
        public int Id { get; set; }

        public int UserId { get; set; }

        public int ScreeningId { get; set; }

        public decimal Price { get; set; }

        public int PromotionId { get; set; }

        public virtual Screening Screening { get; set; } = null!;

        public virtual Promotion Promotion { get; set; } = null!;
    }
}

